code/common/eodtime.q

```diff
@@ -17,16 +17,11 @@ adjtime:{[p]
      :exec adjustment from .tz.t asof `timezoneID`gmtDateTime!(.eodtime.rolltimezone;.z.p);
      };
 
-// function to obtain next logging period
-getperiod:{[p;np;cp]
-     z:rolltimeoffset-adjtime[p];                           // convert rolltimeoffset from rolltimezone to UTC
-     z:`timespan$(mod) . "j"$z, 1D;                         // force time adjust to be between 0D and 1D
-     cp+$[z <= p;z+np;z]                                    // if past time already today, make it next period
-     };
-
 // function to get time (in UTC) of next roll after UTC timestamp, p
 getroll:{[p]
-     getperiod[p;1D;"d"$p]
+     z:rolltimeoffset-adjtime[p];                           // convert rolltimeoffset from rolltimezone to UTC
+     z:`timespan$(mod) . "j"$z, 1D;                         // force time adjust to be between 0D and 1D
+     ("d"$p) + $[z <= p;z+1D;z]                             // if past time already today, make it tomorrow
```

Comment:
Can remove the conditional by using the comparison to determine whether to add a day:
```suggestion
     ("d"$p) + $[z <= p;z+1D;z]                             // if past time already today, make it tomorrow
     ("d"$p)+1D*z<=p                                        // if past time already today, make it tomorrow
```